### DISCUSSION TITLE
FileSearch in a Merge Module?

### DISCUSSION THREAD

**Discussion by MatthiasRInst** [wixtoolset/issues#8546] [Questions]
Hello,

Using WiX 4.0.5, I try to implement a FileSearch in a MergeModule.

<Wix xmlns="http://wixtoolset.org/schemas/v4/wxs">
  <Module Id="FileSearchTest" Codepage="1252" Language="0" Version="19.0.0.0"
          Guid="{2937D893-6654-47A5-BE19-5E2BBAFA1795}" InstallerVersion="500">
    <SummaryInformation Description="FileSearchTest" Keywords="MergeModule,MSI,Database"
                        Manufacturer="MyCompany" />

    <Property Id="FILEFOUND" Secure="yes">
      <DirectorySearch Id="Windows" Path="[WindowsFolder]" >
        <FileSearch Id="MyFile" Name="MyFile.ini" />
      </DirectorySearch>
    </Property>
  </Module>
</Wix>

This gives me the error :
	WIX0001	System.InvalidOperationException: The value '' is not a legal identifier and therefore cannot be modularized.		C:\MyPath\Installation\MergeModules\FileSearch\wix.exe	1	

What is going wrong here?
In WiX 3, this code compiled.

Best regards

Matthias Reuss

**Comment by barnson** [comment]
Exceptions are bugs. Please open an issue with repro repo.

**Comment by Prakash08061993** [comment]
> Hello,
> 
> Using WiX 4.0.5, I try to implement a FileSearch in a MergeModule.
> 
> ```
> <Property Id="FILEFOUND" Secure="yes">
>   <DirectorySearch Id="Windows" Path="[WindowsFolder]" >
>     <FileSearch Id="MyFile" Name="MyFile.ini" />
>   </DirectorySearch>
> </Property>
> ```
> 
> This gives me the error : WIX0001 System.InvalidOperationException: The value '' is not a legal identifier and therefore cannot be modularized. C:\WorkspaceTFS\MasterMagnoliaMigrateWiX\Installation\MergeModules\FileSearch\wix.exe 1
> 
> What is going wrong here? In WiX 3, this code compiled.
> 
> Best regards
> 
> Matthias Reuss

I have had the same issue. I raised the similar question in stackoverflow.
[https://stackoverflow.com/questions/78524594/why-filesearch-when-used-under-module-is-not-working-wix-toolset-v5](url)

**Reply by MatthiasRInst** [reply]
see issue #8558.
